pkg/kubeapiserver/admission/initializer.go

```diff
@@ -136,18 +131,10 @@ func (i *PluginInitializer) Initialize(plugin admission.Interface) {
 		wants.SetInternalKubeClientSet(i.internalClient)
 	}
 
-	if wants, ok := plugin.(admissioninit.WantsExternalKubeClientSet); ok {
-		wants.SetExternalKubeClientSet(i.externalClient)
-	}
-
 	if wants, ok := plugin.(WantsInternalKubeInformerFactory); ok {
 		wants.SetInternalKubeInformerFactory(i.informers)
 	}
 
-	if wants, ok := plugin.(admissioninit.WantsAuthorizer); ok {
-		wants.SetAuthorizer(i.authorizer)
```

Comment:
The PSP plugin needs the authorizer: https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/security/podsecuritypolicy/admission.go#L70

Reply:
I see, the authorizer and externalClient is set by the generic apiserver's plugin initializer.

Reply:
yeah, that's the idea. Move the `authorizer` and `externalClient` down to `apiserver`